sdk/eventhub/azure-eventhub/azure/eventhub/aio/_eventprocessor/event_processor.py

```diff
@@ -129,6 +129,8 @@ def __repr__(self) -> str:
                     self._id,
                     partition_id
                 )
+                if partition_id not in self._consumers:  # task is cancelled before the consumer is created
+                    del self._tasks[partition_id]
```

Comment:
Adding more context here to illustrate the usage of these two-lines code:

When we stop an EventHubConsumer(EventProcessor) in the async world, chances are that an async task (the coroutine method `_receive`) is cancelled before it's executed. In this case, the program would hang when `EventProcessor.stop()` is called. -- The task never gets a chance to be removed from the dict.

The self._consumers[partition_id] serves as a flag whether  `_receive` has been executed.
If it's cancelled while the consumer is not set, then we should proactively remove the task here as the `_receive` doesn't get a chance to execute itself.
If it's cancelled while the consumer is already set, then the coroutine method (`receive`) itself will be responsible for removing the task.

Reply:
Good explanation. Thanks Adam.